pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/PinotFSFactory.java

```diff
@@ -74,9 +84,21 @@ public static void init(PinotConfiguration fsFactoryConfig) {
   }
 
   public static PinotFS create(String scheme) {
-    PinotFS pinotFS = PINOT_FS_MAP.get(scheme);
-    Preconditions.checkState(pinotFS != null, "PinotFS for scheme: %s has not been initialized", scheme);
-    return pinotFS;
+    PinotFSInfo pinotFSInfo = PINOT_FS_MAP.get(scheme);
+    Preconditions.checkState(pinotFSInfo != null, "PinotFS for scheme: %s has not been initialized", scheme);
+    return pinotFSInfo.getPinotFS();
+  }
+
+  public static PinotFS createNewInstance(String scheme) {
+    PinotFSInfo pinotFSInfo = PINOT_FS_MAP.get(scheme);
+    Preconditions.checkState(pinotFSInfo != null, "PinotFS for scheme: %s has not been initialized", scheme);
```

Comment:
perhaps make the error msg "FS scheme: %s has not been `registered`" to be more specific